lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1027,18 +1032,29 @@ def view_init(self, elev=None, azim=None, roll=None, vertical_axis="z"):
             dict(x=0, y=1, z=2), vertical_axis=vertical_axis
         )
 
-    def set_proj_type(self, proj_type):
+    def set_proj_type(self, proj_type, focal_length=None):
         """
         Set the projection type.
 
         Parameters
         ----------
         proj_type : {'persp', 'ortho'}
-        """
-        self._projection = _api.check_getitem({
-            'persp': proj3d.persp_transformation,
-            'ortho': proj3d.ortho_transformation,
-        }, proj_type=proj_type)
+            The projection type.
+        focal_length : float, default: None
+            For a projection type of 'persp', the focal length of the virtual
+            camera. Must be >0. If None, defaults to 1.
+            The focal length can be computed from a desired Field Of View via
+            the equation: focal_length = 1/tan(FOV/2)
+        """
+        if proj_type == 'persp':
+            self.focal_length = focal_length
+            if focal_length is None:
+                self.focal_length = 1
+        elif proj_type == 'ortho':
+            self.focal_length = np.inf
```

Comment:
not a fan of dropping user input on the floor here.

Reply:
Added a check